### ISSUE TITLE
add waffle switch to enable mv3 validation in linter

### ISSUE THREAD

**Issue by eviljeff** [mozilla/addons-server#17450]
Experimental support for mv3 was added in mozilla/addons-linter#3721 but we need to pass `max-manifest-version=3` to enable it.  

Behind a waffle switch because it's still experimental and until Firefox support is more advanced we don't want mv3 extensions uploaded to AMO right now even if they _do_ have a valid manifest.

**Comment by willdurand** [comment]
cc @rpl

**Comment by willdurand** [comment]
@rpl when do you want to be able to submit MV3 extensions on AMO **-dev**?

**Comment by eviljeff** [comment]
mv3 submission will likely be available on dev shortly after the waffle switch commit has landed (pretty soon too I assume as the PR is uncomplicated).  QA will have to toggle it off and on to verify this issue.  Though it won't be particularly useful to do that until more work is done.

**Comment by rpl** [comment]
> @rpl when do you want to be able to submit MV3 extensions on AMO **-dev**?

@willdurand No specific timing preference from my side at the moment. I don't think any of the MV3 work on the Firefox side currently in progress is blocked on being able to sign an MV3 on AMO -dev.

At the moment I think that having it on -dev may help mainly to QA verify the changes we do on the addons-linter that are also MV3-related also along with its integration with AMO (instead of just through its command line interface).

(I did also mention this during the team standup today, to make the rest of the team aware of this and in case they had a specific need for this to happen with a specific timing, which doesn't seem to be the case based on the quick chat we had).

**Comment by ioanarusiczki** [comment]
@eviljeff I see the switch, I enabled it and started testing addon submissions using dev hub. I'm going to compare validation results switch disabled versus switch enabled

**Comment by ioanarusiczki** [comment]
@eviljeff 

With `enable-mv3-submissions` switch enabled I tested several addon submissions (.zip and .xpi formats).
-  I focused more on the listed channel but also tried some unlisted submissions.
- I uploaded some extensions but also a static theme, lang pack and dict and new versions - looks good. 
- I tried some linter errors (mainly following the model from https://github.com/mozilla/addons-linter/issues/3721#issuecomment-839869503 ) and I did not see anything different than before. 
- in case I did not cover the issue let me know on what should I focus on.
